src/libraries/System.Private.CoreLib/src/System/Globalization/CompareInfo.cs

```diff
@@ -825,10 +825,108 @@ public bool IsPrefix(ReadOnlySpan<char> source, ReadOnlySpan<char> prefix, Compa
             return source.StartsWithOrdinalIgnoreCase(prefix);
         }
 
-        private unsafe bool StartsWithCore(ReadOnlySpan<char> source, ReadOnlySpan<char> prefix, CompareOptions options) =>
+        /// <summary>
+        /// Determines whether a string starts with a specific prefix.
+        /// </summary>
+        /// <param name="source">The string to search within.</param>
+        /// <param name="prefix">The prefix to attempt to match at the start of <paramref name="source"/>.</param>
+        /// <param name="options">The <see cref="CompareOptions"/> to use during the match.</param>
+        /// <param name="matchLength">When this method returns, contains the number of characters of
+        /// <paramref name="source"/> that matched the desired prefix. This may be different than the
+        /// length of <paramref name="prefix"/> if a linguistic comparison is performed. Set to 0
+        /// if the prefix did not match.</param>
+        /// <returns>
+        /// <see langword="true"/> if <paramref name="prefix"/> occurs at the start of <paramref name="source"/>;
+        /// otherwise, <see langword="false"/>.
+        /// </returns>
+        /// <exception cref="ArgumentException">
+        /// <paramref name="options"/> contains an unsupported combination of flags.
+        /// </exception>
+        /// <remarks>
+        /// This method has greater overhead than other <see cref="IsPrefix"/> overloads which don't
+        /// take a <paramref name="matchLength"/> argument. Call this overload only if you require
+        /// the match length information.
+        /// </remarks>
+        public unsafe bool IsPrefix(ReadOnlySpan<char> source, ReadOnlySpan<char> prefix, CompareOptions options, out int matchLength)
+        {
+            // Just like the existing IsPrefix overloads, we'll allow an early
+            // exit here before 'options' is validated.
+
+            if (prefix.IsEmpty)
+            {
+                matchLength = 0;
```

Comment:
>matchLength [](start = 16, length = 11)

is this correct? I think we need to get how many sort ignorable characters and return that.

Reply:
It's a legal return value. This matches the existing logic already in `string.IndexOf(..., int*, ...)`, which states that a zero-length value always has a zero-length match within the target string.

https://github.com/dotnet/runtime/blob/6fb6750c5615279a3773f40ee2360abdcbb988d5/src/libraries/System.Private.CoreLib/src/System/Globalization/CompareInfo.cs#L1206-L1223

See also the comment [which begins here](https://github.com/dotnet/runtime/blob/aa5fdab9654d74bc6274c0b5d820272c8e859621/src/libraries/System.Private.CoreLib/src/System/String.Searching.cs#L223), which says that zero-weight code points may or may not be consumed by searching logic.

Reply:
I'll add some clarifying comments to the docs based on this.

Reply:
I am fine with that. I wanted this be explicit and documented. Although this behavior not the best (from my point of view) but it is acceptable and not worth it for this corner case especially the perf maybe more desirable.

Reply:
See https://github.com/dotnet/dotnet-api-docs/issues/4676 for some clarification.

Reply:
thanks @GrabYourPitchforks